src/Adapter/MSTest.CoreAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -114,10 +114,21 @@ internal TestCase ToTestCase()
                 this.TestMethod.FullClassName,
                 this.TestMethod.Name);
 
-            TestCase testCase = new TestCase(fullName, TestAdapter.Constants.ExecutorUri, this.TestMethod.AssemblyName);
-            testCase.DisplayName = string.IsNullOrEmpty(this.DisplayName) ? this.TestMethod.Name : this.DisplayName;
+            TestCase testCase;
 
+            if (this.TestMethod.HasManagedMethodAndType)
+            {
+                testCase = new TestCase(fullName, this.TestMethod.ManagedType, this.TestMethod.ManagedMethod, TestAdapter.Constants.ExecutorUri, this.TestMethod.AssemblyName);
```

Comment:
@Haplois Won't this fail If the version of mstest adapter that includes the current FQN changes is loaded in VS 2017 where the TP OM that will be loaded will not contain this constructor?

Reply:
It will fail. I am moving these changes to a separate assembly and NuGet package. I'll have a PR soon.